src/components/pages/profile-overview/profile-overview.tsx

```diff
@@ -19,11 +19,9 @@ import {
 } from "@src/utils/format";
 
 const ProfileOverview: React.FC<{
-  data?: {
-    userProfile: Profile;
-    userAddresses: Address[];
-  };
-}> = ({ data }) => {
+  userProfile?: Profile;
+  userAddresses?: Address[];
```

Comment:
```suggestion
  userProfile?: Profile;
  userAddresses: Address[];
```

`Profile` can be undefined if the user doesn't have any, but `userAddresses` shouldn't be because when we send the data from the [API page](https://github.com/fewlinesco/connect-account/blob/d1cf590c181456b51d4b00e5ae418b9f656329bf/src/pages/api/profile/addresses/index.ts#L85) (which comes from Connect.Profile), if there's no address, it's an empty array

Reply:
I think the possibly undefined here is more about SWR data fetching way than Profile possible response. I'm gonna test it to see.

Reply:
Yep, I confirm. The possibly `undefined` here is just here because we fetch data client side with SWR. 

<img width="889" alt="image" src="https://user-images.githubusercontent.com/50053259/119653072-31985180-be27-11eb-900c-8e04bde73e49.png">

Reply:
Yes, but when passing the data to the component here: https://github.com/fewlinesco/connect-account/blob/8a8fff5c6bc33f2c0a9fc783b05643c60ba3ef4b/src/pages/account/profile/index.tsx#L58

Maybe we could have a default value to `[]` in case of undefined here: https://github.com/fewlinesco/connect-account/blob/8a8fff5c6bc33f2c0a9fc783b05643c60ba3ef4b/src/components/pages/profile-overview/profile-overview.tsx#L23

Reply:
Yep. And how should we do for the Skeleton component if we have a default value equal to `[]` then ? I mean, we couldn't rely on the fact that `userAddresses` is an empty array, because it could mean in this case that either data isn't ready yet or that there isn't any data at all.

Reply:
Actually, if we don't have any address, that means we:
- display de skeleton component until we get the response
- remove it (because no address)
- display a button

And it feels weird too don't you think?

Reply:
I feel this is just a naming problem. This component receives data from a SWR call (from what I understand) which has two values for the data received: `undefined` and what we type it with. 

Which means that we don't pass to this component the data really but a "state of the data" (or a representation of the data). I feel there's two way to reconcile your two points of view, either renaming `userAddresses` into something like `userAddressesData` and agreeing that it's not yet the real data (which could only be an array) or else have the component receive an object that could be typed something like this:

```ts
type UserProfileData = "loading" | { userProfile: Profile; userAddresses: Address[] }
```

(the `"loading"` part could be an object with something to mark that it's in a loading state)

That way, it's only when the component received a loading state that we display the skeletton, not when we have the data (whateter form it has).

What do you think?

Reply:
> What do you think?

👇 

> (the "loading" part could be an object with something to mark that it's in a loading state)

I totally agree with that

> renaming userAddresses into something like userAddressesData and agreeing that it's not yet the real

Less with that part because I don't think the address component should treat with undefined data because this specific data can't be (not like the Profile one)

Reply:
But if we go on the `loading` road, this should be done in this PR though

Reply:
>(the "loading" part could be an object with something to mark that it's in a loading state)

I really like this one too. 

I guess I'm gonna do a task for that then 🙂

Reply:
Here's an exemple of what I had in mind https://www.typescriptlang.org/play?#code/C4TwDgpgBACgTgewGYEsA20C8UDeVgIEBcUAjFAL4BQVokUAqgM4RzzLoQAiAhsD1Gx4m-YBBIAiNAh4ATFADsA5hMpQAPrigi+4qFJmyIsiQBoosvjxLtUGSgG4aAYwQKRUAK4s2iO9ysSZlZbTl5+QSgAWT4ACwA6OB4FWQQAWwAKAEooAB4oAAZ4gFYoAH4tHTFJaTlFFQooEhwqvQM5YzNLfmb8QgQSUgpqKld3YHwID2wM7xC-MKt41sFMbHb5ZQkcqig9-b2KhU80NChdg-2SOd8ODHCeeO7HggIqIA